internal/pkg/task/env_runner.go

```diff
@@ -76,6 +79,7 @@ func (r *EnvRunner) Run() ([]*Task, error) {
 	if err != nil {
 		return nil, fmt.Errorf(fmtErrSecurityGroupsFromEnv, r.Env, err)
 	}
+	securityGroups = append(securityGroups, r.SecurityGroups...)
```

Comment:
What do you think about checking the number of security groups given the limit of 5?
```go
if total := len(securityGroups) + len(r.SecurityGroups); total > 5 {
   return  nil, fmt.Errorf("unable to run task with more than 5 security groups: (%d) %s", total, strings.Join(append(securityGroups, r.SecurityGroups...), ","))
}
securityGroups = append(securityGroups, r.SecurityGroups...)
```

Reply:
oh good call. I was also thinking that maybe we should dedup the list of SGs here, since I think `--generate-cmd` output is going to include the same core security group that `--app` and `--env` are going to include based on running inside a copilot environment.